tests/unit/test_worker.py

```diff
@@ -385,6 +385,7 @@ def job_func(a, b):  # pylint: disable=unused-argument
         task_name="job",
         queue="yay",
     )
+    app.job_manager.get_job_status_async = mocker.AsyncMock(return_value="doing")
```

Comment:
Just curious: what happens if we don't mock? The InMemory implementation doesn't answer correctly ?

Reply:
It'll throw. The job is not put into the worker queue. The `run_job` method is tested in isolation.

Reply:
Ah, makes sense!